# --- T2-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# 
# T2 SDE: package/.../uclibc/hotfix-genwctype.patch
# Copyright (C) 2005 The T2 SDE Project
# 
# More information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- T2-COPYRIGHT-NOTE-END ---

Fix uninitialized stack variable, causes a segfault with O2
optimization and gcc 4.0.2.
     Michael Tross

Nur in uClibc-0.9.28.work/extra/locale: gen_wctype.
diff -ur uClibc-0.9.28/extra/locale/gen_wctype.c uClibc-0.9.28.work/extra/locale/gen_wctype.c
--- uClibc-0.9.28/extra/locale/gen_wctype.c	2005-08-18 00:49:41.000000000 +0200
+++ uClibc-0.9.28.work/extra/locale/gen_wctype.c	2005-12-10 07:35:46.000000000 +0100
@@ -10,10 +10,6 @@
 #include <wchar.h>
 #include <ctype.h>
 
-#ifdef __linux__
-#include <sys/resource.h>
-#endif
-
 #ifndef _CTYPE_H
 #define _CTYPE_H
 #endif
@@ -247,14 +243,6 @@
 	static const char empty_slot[] = "empty_slot";
 	int built = 0;
 
-#ifdef __linux__
-	struct rlimit limit;
-
-	limit.rlim_max = RLIM_INFINITY;
-	limit.rlim_cur = RLIM_INFINITY;
-	setrlimit(RLIMIT_STACK, &limit);
-#endif
-
 #define INIT_TYPENAME(X) typename[__CTYPE_##X] = "C_" #X
 
 	for (i=0 ; i < 16 ; i++) {
@@ -796,6 +784,7 @@
 		ti[i] = ut + j;
 		j += blocksize;
 	}
+	memset(uniqblock, 0, sizeof(uniqblock));
 
 	/* sort */
 	nu_val = blocksize;
Nur in uClibc-0.9.28.work/extra/locale: wctables.h.
